dart/collision/fcl/FCLCollisionDetector.h

```diff
@@ -38,50 +38,104 @@
 #ifndef DART_COLLISION_FCL_FCLCOLLISIONDETECTOR_H_
 #define DART_COLLISION_FCL_FCLCOLLISIONDETECTOR_H_
 
+#include <vector>
 #include <fcl/collision_object.h>
-#include <fcl/collision_data.h>
-#include <fcl/broadphase/broadphase.h>
-
 #include "dart/collision/CollisionDetector.h"
 
 namespace dart {
 namespace collision {
 
-class FCLCollisionNode;
+class FCLCollisionObject;
 
-/// FCLCollisionDetector
 class FCLCollisionDetector : public CollisionDetector
 {
 public:
-  /// Constructor
-  FCLCollisionDetector();
 
-  /// Destructor
+  static std::shared_ptr<FCLCollisionDetector> create();
+
+  /// Constructor
   virtual ~FCLCollisionDetector();
 
+  /// Return engine type "FCL"
+  static const std::string& getTypeStatic();
+
   // Documentation inherited
-  virtual bool detectCollision(bool _checkAllCollisions,
-                               bool _calculateContactPoints) override;
+  const std::string& getType() const override;
 
   // Documentation inherited
-  virtual CollisionNode* createCollisionNode(dynamics::BodyNode* _bodyNode)
-  override;
+  std::shared_ptr<CollisionGroup> createCollisionGroup() override;
 
-  /// Get collision node given FCL collision geometry
-  CollisionNode* findCollisionNode(
-      const fcl::CollisionGeometry* _fclCollGeom) const;
+  // Documentation inherited
+  bool detect(CollisionGroup* group,
+              const Option& option, Result& result) override;
 
-  /// Get collision node given FCL collision object
-  FCLCollisionNode* findCollisionNode(
-      const fcl::CollisionObject* _fclCollObj) const;
+  // Documentation inherited
+  bool detect(CollisionGroup* group1, CollisionGroup* group2,
+              const Option& option, Result& result) override;
+
+  /// FCL's primitive shape support is still not complete. FCL 0.4.0 fixed lots
+  /// of the bugs, but it still returns single contact point per shape pair
+  /// except for box-box collision. For this reason, we recommend using mesh for
+  /// primitive shapes until FCL fully support primitive shapes.
+  enum PrimitiveShape_t
```

Comment:
You should make it clear in this comment that primitive shape support is only buggy for *contact point* computation. It works fine for binary checks.

Also, I suggest making this an `enum class` and renaming it to `PrimitiveShape`. Why does this have the `_t` suffix?